File: TimetableGeneratorEngine/InputDataGenerator/ClassHour.cpp
#include "stdafx.h"
#include "ClassHour.h"

json ClassHour::GetJson()
{
    json res;
    res["_id"] = m_id;
    res["number"] = m_number;
    res["class"] = m_classs.GetJson();
    res["subject"] = m_subject.GetJson();
    res["teacher"] = m_teacher.GetJson();
    res["weight"] = m_weight;
    return res;
}

void ClassHour::ChangeSubjectLocations(const std::vector<Location>& p_locations)
{
    m_subject.SetLocations(p_locations);
}

File: TimetableGenerator/backend/src/models/ClassHour.ts
import mongoose from 'mongoose'

const schema = new mongoose.Schema({
  number: Number,

  class: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: "Class" 
  },

  subject: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: "Subject" 
  },

  teacher: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: "Teacher" 
  },

  weight: Number
});

export default mongoose.model('ClassHour', schema);